kubespawner/spawner.py

```diff
@@ -2663,6 +2666,12 @@ async def _start(self):
         if self.enable_user_namespaces:
             await self._ensure_namespace()
 
+        # namespace can be changed via kubespawner_override, start watching pods only after
+        # load_user_options() is called
+        await self._start_watching_pods()
+        if self.events_enabled:
+            await self._start_watching_events()
```

Comment:
Previously we started up the reflectors concurrently, and waited for the startup when needed at a later time.

```python
        self._start_watching_pods()
        if self.events_enabled:
            self._start_watching_events()
```

We can at least do a slight improvement from waiting directly on A and then B, but starting both A and B and then waiting for both.

```suggestion
        start_futures = [self._start_watching_pods()]
        if self.events_enabled:
            start_futures.append(self._start_watching_events())
        await asyncio.wait(start_futures)
```

Reply:
I note that I used `asyncio.wait` as compared to `asyncio.gather`, they are very similar I think but I'm not sure if either this code block should use `asyncio.gather` or the one in `_stop_all_reflectors` should use `asyncio.wait` - or if it doesn't matter much at all.

I understand that `.wait` doesn't care about the return values, which we don't here, but `.gather` does.

Reply:
According to documentation:
https://docs.python.org/3/library/asyncio-task.html#asyncio.gather
https://docs.python.org/3/library/asyncio-task.html#asyncio.wait

`wait` does not propagate exception from a coroutine, but `gather` does:
```
If return_exceptions is False (default), the first raised exception is immediately propagated to the task that awaits on gather(). Other awaitables in the aws sequence won’t be cancelled and will continue to run.
```

So I used `gather` to propagate raised exception to the spawner, but with some wrapper to cancel tasks which are not done yet.